conda_lock/models/pip_repository.py

```diff
@@ -19,11 +32,24 @@ def base_url(self) -> str:
         full_url = urlparse(self.url)
         return full_url.scheme + "://" + full_url.netloc
 
+    @property
+    def expanded_basic_auth(self) -> Optional[UsernamePasswordDict]:
+        parsed_url = urlparse(self.url)
+        if parsed_url.username is None:
+            return None
+        username = expandvars(parsed_url.username)
+        password = expandvars(parsed_url.password or "")
+        return UsernamePasswordDict(username=username, password=password)
+
     @property
     def stripped_base_url(self) -> str:
         """The base URL of the pip repository, without any basic auth."""
-        base_url = urlparse(self.base_url)
-        return urlunparse(base_url._replace(netloc=base_url.netloc.split("@", 1)[-1]))
+        return stripped_url(self.base_url)
+
+    @property
```

Comment:
nit: stripped_base_url and stripped_url are the same thing. Do we need/want both?

Reply:
I don't think they are the same. Should have added example values. :see_no_evil:

Reply:
oh right, sorry, I stupidely missed base_url vs url in the arg :(